table_helper.cc

```diff
@@ -69,9 +69,10 @@ future<> table_helper::setup_table(cql3::query_processor& qp, service::migration
 }
 
 future<bool> table_helper::try_prepare(bool fallback, cql3::query_processor& qp, service::query_state& qs) {
-    auto& stmt = fallback ? _insert_cql_fallback : _insert_cql;
+    // Note: `_insert_cql_fallback` is known to be engaged if `fallback` is true, see cache_table_info below.
+    auto& stmt = fallback ? _insert_cql_fallback.value() : _insert_cql;
     try {
-        shared_ptr<cql_transport::messages::result_message::prepared> msg_ptr = co_await qp.prepare(_insert_cql, qs.get_client_state());
+        shared_ptr<cql_transport::messages::result_message::prepared> msg_ptr = co_await qp.prepare(stmt, qs.get_client_state());
```

Comment:
The fix was merged into the wrong commit, this hunk belongs to the previous one.

Reply:
ouch, fixed